src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -310,10 +660,10 @@ public void startADBatchTask(
                     internalListener.onFailure(e);
                 }
             });
-            listener.onResponse(new ADBatchAnomalyResultResponse(clusterService.localNode().getId(), runTaskRemotely));
+            delegatedListener.onResponse(new ADBatchAnomalyResultResponse(clusterService.localNode().getId(), runTaskRemotely));
```

Comment:
Can we rename executeADBatchTask to executeADBatchTaskOnWorkerNode for better readability?

Reply:
Will fix in new PR